includes/models/model.llms.order.php

```diff
@@ -1436,7 +1425,27 @@ public function init( $person, $plan, $gateway, $coupon = false ) {
 			$this->set( 'access_period', $plan->get( 'access_period' ) );
 		}
 
-		do_action( 'lifterlms_new_pending_order', $this, $person );
+		/**
+		 * Action triggered after a the order is initialized.
+		 *
+		 * @since Unknown.
+		 * @since [version] Added `$user_data` parameter.
+		 *                 The `$student` parameter returns an "empty" student object
+		 *                 if the method's input data is an array instead of an existing
+		 *                 user object.
+		 *
+		 * @param LLMS_Order   $order                           The order object.
+		 * @param LLMS_Student $student                         The student object. If an array of data is passed
+		 *                                                      to `LLMS_Order::init()` then an empty student object
+		 *                                                      will be passed.
+		 * @param array|LLMS_Student|WP_User|integer $user_data User data.
+		 */
+		do_action(
+			'lifterlms_new_pending_order',
+			$this,
+			is_array( $user_data ) ? new LLMS_Student( -1 ) : llms_get_student( $user_data ),
```

Comment:
perplexed about this `-1` considering that it will populate the `LLMS_Student::$id`, and `LLMS_Abstract_User_Data::get_user_id()` returns and `absint($id)`, so it'll return `1` => usually the first admin mmm.

Reply:
good catch... `-1` ensures the `WP_User` object isn't found. There's no way to construct an empty object otherwise as the current user ID would be used by default.

I'll have to do some refactoring on the student class to make that work the way I wanted it to...

Really this is for backwards compatibility on the hook so maybe a better option would be to just deprecate the filter and add a replacement that functions differently

Reply:
Addressed in 1666a47d08928dbecdc06db7e679308435ba355a

I think this is a more sane way of accomplishing this. Thoughts?